seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-sftp-e2e/src/test/java/org/apache/seatunnel/e2e/connector/file/fstp/SftpFileIT.java

```diff
@@ -0,0 +1,208 @@
+        Container.ExecResult excelProjectionReadResult =
+                container.executeJob("/excel/sftp_excel_projection_to_assert.conf");
+        Assertions.assertEquals(
+                0, excelProjectionReadResult.getExitCode(), excelProjectionReadResult.getStderr());
+        // test write sftp text file
+        Container.ExecResult textWriteResult =
+                container.executeJob("/text/fake_to_sftp_file_text.conf");
+        Assertions.assertEquals(0, textWriteResult.getExitCode());
+        // test read skip header
+        Container.ExecResult textWriteAndSkipResult =
+                container.executeJob("/text/sftp_file_text_skip_headers.conf");
+        Assertions.assertEquals(0, textWriteAndSkipResult.getExitCode());
+        // test read sftp text file
+        Container.ExecResult textReadResult =
+                container.executeJob("/text/sftp_file_text_to_assert.conf");
+        Assertions.assertEquals(0, textReadResult.getExitCode());
+        // test read sftp text file with projection
+        Container.ExecResult textProjectionResult =
+                container.executeJob("/text/sftp_file_text_projection_to_assert.conf");
+        Assertions.assertEquals(0, textProjectionResult.getExitCode());
+        // test write sftp json file
+        Container.ExecResult jsonWriteResult =
+                container.executeJob("/json/fake_to_sftp_file_json.conf");
+        Assertions.assertEquals(0, jsonWriteResult.getExitCode());
+        // test read sftp json file
+        Container.ExecResult jsonReadResult =
+                container.executeJob("/json/sftp_file_json_to_assert.conf");
+        Assertions.assertEquals(0, jsonReadResult.getExitCode());
+    }
+
+    public static void initDate(String host, int mappedPort) {
+        Map<String, String> filesMap = new HashMap<>();
+        filesMap.put(
+                "/excel/e2e.xlsx", "/tmp/seatunnel/read/excel/name=tyrantlucifer/hobby=coding/");
+        filesMap.put("/json/e2e.json", "/tmp/seatunnel/read/json/name=tyrantlucifer/hobby=coding/");
+        filesMap.put("/text/e2e.txt", "/tmp/seatunnel/read/text/name=tyrantlucifer/hobby=coding/");
+
+        for (Map.Entry<String, String> entry : filesMap.entrySet()) {
+            String localPath = entry.getKey();
+            String remotePath = entry.getValue();
+            Path path = ContainerUtil.getResourcesFile(localPath).toPath();
+
+            tmpFileToSftpServerWithCustomPath(
+                    host, mappedPort, USERNAME, PASSWORD, path.toFile(), remotePath);
+        }
+    }
+
+    @SneakyThrows
+    public static void tmpFileToSftpServerWithCustomPath(
+            String host,
+            int port,
+            String username,
+            String password,
+            File localFilePath,
+            String remotePath) {
+        JSch jsch = new JSch();
+        Session session = null;
+        ChannelSftp channelSftp = null;
+
+        try {
```

Comment:
Use `@SneakyThrows` why use try?

Reply:
@TyrantLucifer 
try finally It is to ensure the release of links.
@SneakyThrows It is to throw an exception instead of the following logic:
try{
}catch(Exception e){
throw new RuntimeException(e);
}

Reply:
Of course, if you think it's not good to write like this, I can change it to:
try{
}catch(Exception e){
throw new RuntimeException(e);
} finally {
.....
}

Reply:
@TyrantLucifer 
try finally It is to ensure the release of links.
@SneakyThrows It is to throw an exception instead of the following logic:
try{
}catch(Exception e){
throw new RuntimeException(e);
}

Decompiled code：
try{
  try{
}finally{
}
}catch(Exception e){
throw new RuntimeException(e);
}

Reply:
After thinking about it, this may not be a good way, but I changed it:
  try{
  } catch (Exception e) {
            throw new RuntimeException(e);
        } finally {
}

Reply:
One more question: Why not copy file using copy but using this method? I think it's not a grace solution.

Reply:
@TyrantLucifer 
<img width="925" alt="图片" src="https://user-images.githubusercontent.com/60029759/234474288-ea5e61b3-8f81-4ecf-9917-bdb09b281fd5.png">
You're talking about the method shown in the image above, right? Copying the files into the container. I've tried this method before, but when I connected to SFTP, I couldn't find those files. After researching relevant information, I found that I need to write a program to upload the files to the container's SFTP service directory, which is how the current code logic came about.

Reply:
On this issue, I also asked chgpt for help at that time：
<img width="934" alt="图片" src="https://user-images.githubusercontent.com/60029759/234476277-98a61cfb-2d38-47a5-bcfd-5c75fe921b7b.png">

Reply:
SFTP Protocol use the linux user to mapping file driectory permission, for example `docker run -p 22:22 -d atmoz/sftp foo:pass:::upload` container will create a linux user `foo` and create `/home/foo/upload` to enable you upload files in it. So in this case, you can upload files into `/home/foo/tmp`. BTW, `foo` is not a good choice, `seatunnel` or `st` will be better.
 
![image](https://user-images.githubusercontent.com/51053924/234530871-e37142dc-f04f-47db-887a-3c234907dc36.png)

Reply:
<img width="761" alt="图片" src="https://user-images.githubusercontent.com/60029759/234540272-35ef444f-c407-44a6-882a-26f64a6bff54.png">
I use foo:pass:::tmp mode to initialize the container, but withcopyFileToContainer also places the content in tmp/xxx/xxx and finds no content.

Reply:
![image](https://user-images.githubusercontent.com/51053924/234544753-a07b255c-b5c6-44fa-864e-604f9e4b1432.png)

Reply:
Got it. I'll revise it and submit it.